aws-datastore/src/test/java/com/amplifyframework/datastore/syncengine/MergerTest.kt

```diff
@@ -667,6 +668,61 @@ class MergerTest {
         assertEquals(expectedStorageItemChanges, capturedStorageItemChanges)
     }
 
+    /**
+     * THIS IS NOT FINAL
+     * When several models with the same identifier, but different versions with different flag for deleted,
+     * Only the model with the latest version should be considered as it represents the latest known state of the model
+     * @throws
+     * @throws
+     */
+    @Test
+    @Throws(DataStoreException::class, InterruptedException::class)
+    fun temporaryMergerTestReconcilation() {
+
+        var capturedStorageItemChanges = mutableListOf<StorageItemChange.Type>()
+        val changeTypeConsumer = Consumer<StorageItemChange.Type> {
+            capturedStorageItemChanges.add(it)
+        }
+
+        // Random UUID following RFC 4122 version 4 spec
+        val sameRandomId = UUID.randomUUID().toString()
+
+        val remotemodels = listOf(
+            ModelWithMetadata(
+                Blog.builder().name("Hideo K.").id("DS1").build(),
+                ModelMetadata(sameRandomId, false, 1, Temporal.Timestamp.now())
+            ),
+            ModelWithMetadata(
+                Blog.builder().name("Hideo K.").id("DS1").build(),
+                ModelMetadata(sameRandomId, true, 3, Temporal.Timestamp.now())
+            ),
+            ModelWithMetadata(
+                Blog.builder().name("Hideo K.").id("DS1").build(),
+                ModelMetadata(sameRandomId, true, 2, Temporal.Timestamp.now())
+            ),
+            ModelWithMetadata(
+                Blog.builder().name("Hideo K.").id("DS2").build(),
+                ModelMetadata(sameRandomId, false, 11, Temporal.Timestamp.now())
+            ),
+            ModelWithMetadata(
+                Blog.builder().name("Hideo K.").id("DS2").build(),
+                ModelMetadata(sameRandomId, true, 8, Temporal.Timestamp.now())
+            ),
+            ModelWithMetadata(
+                Blog.builder().name("Hideo K.").id("DS2").build(),
+                ModelMetadata(sameRandomId, true, 17, Temporal.Timestamp.now())
+            ),
+            ModelWithMetadata(
+                Blog.builder().name("Hideo K.").id("DS2").build(),
+                ModelMetadata(sameRandomId, false, 34, Temporal.Timestamp.now())
+            )
+        )
+
+        val observer = merger.merge(remotemodels, changeTypeConsumer).test()
+        assertTrue(observer.await(REASONABLE_WAIT_TIME, TimeUnit.MILLISECONDS))
+        observer.assertNoErrors().assertComplete()
```

Comment:
This is only testing that an error didn't throw. This isn't testing that the duplicate records were removed in the proper order.

Reply:
Added test cases to compare the local models with the expected models